lib/src/main/scala/org/typelevel/scalacoptions/ScalacOptions.scala

```diff
@@ -60,11 +60,25 @@ private[scalacoptions] trait ScalacOptions {
   val feature =
     ScalacOption("-feature", _ => true)
 
+  /** Compile for a specific version of the Java platform. Supported targets: 8, 9, ..., 17, 18.
+    *
+    * The java-output-version flag is supported only on JDK 9 and above, since it relies on the
+    * functionality provided in
+    * [[http://openjdk.java.net/jeps/247 JEP-247: Compile for Older Platform Versions]].
+    */
+  def javaOutputVersion(version: String) =
+    ScalacOption(
+      "-java-output-version",
+      List(version),
+      version => JavaMajorVersion.javaMajorVersion >= 9 && version >= V3_1_2
+    )
+
   /** Compile for a specific version of the Java platform. Supported targets: 8, 9, ..., 17, 18.
     *
     * The release flag is supported only on JDK 9 and above, since it relies on the functionality
     * provided in [[http://openjdk.java.net/jeps/247 JEP-247: Compile for Older Platform Versions]].
     */
+  @deprecated("Use javaOutputVersion instead", "3.1.2")
```

Comment:
This need to be removed and instead the line below:
```
version => JavaMajorVersion.javaMajorVersion >= 9 && version >= V2_12_5
```
needs to be changed to:
```
version => JavaMajorVersion.javaMajorVersion >= 9 && version >= V2_12_5 && version < V3_1_2
```

Reply:
Or maybe not since there is an alias for backcompat?

Reply:
Thank you, I've made those changes.

Reply:
@satorg No rush but maybe we could look into this before cutting a release?

Reply:
Sorry it should be:
```
version => JavaMajorVersion.javaMajorVersion >= 9 && version.between(V2_12_5, V3_1_2)
```
But again not sure if we should disable it yet because that flag is still working. Any opinion @TonioGela @satorg @armanbilge?

Reply:
ok fixed, but I take your point about disabling or not.

Reply:
I think if We disable this for scala > 3.1.2, it may cause some suprise for users as it takes no effect. So, I think We should not disable it.

Reply:
Yeah, let's keep it then. People will get the warning if there is one and we can remove it once the compiler removes it.